src/main/java/com/ericsson/eiffel/remrem/semantics/validator/EiffelValidator.java

```diff
@@ -72,41 +89,127 @@ public EiffelValidator(String schemaResourceName, String eventType, List<String>
             throw new IllegalArgumentException(message, e);
         }
     }
-
-    public void validate(JsonObject jsonObjectInput) throws EiffelValidationException {
+    public JsonObject validate(JsonObject jsonObjectInput) throws EiffelValidationException {
+        return validate(jsonObjectInput, false);
+    }
+    
+    public JsonObject validate(JsonObject jsonObjectInput, Boolean lenientValidation) throws EiffelValidationException {
+        JsonArray remremGenerateFailures = new JsonArray();
         try {
-            ProcessingReport report = validationSchema.validate(JsonLoader.fromString(jsonObjectInput.toString()));
-            if (!report.isSuccess()) {
+            final ProcessingReport report = validationSchema.validate(JsonLoader.fromString(jsonObjectInput.toString()),
+                    true);
+            if (!report.isSuccess() && lenientValidation) {
+                log.debug("Trying to revalidate the json for optional fields");
                 log.warn(jsonObjectInput.toString());
```

Comment:
We shouldn't log this as a warning unless the lenient validation also fails. If lenient validation is enabled it's totally normal to end up on this line.